server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java

```diff
@@ -320,9 +323,10 @@ private PhysicalPlan transformQuery(QueryOperator queryOperator)
                 dataTypeConsistencyChecker.put(measurementChecked, dataType);
               }
 
-              // update measurementSetOfGivenSuffix and measurement location
-              if (measurementSetOfGivenSuffix.add(measurementChecked)) {
-                loc++;
+              // update measurementSetOfGivenSuffix and Normal measurement
```

Comment:
rename the  pathForDataType to pathWithAggregation

Reply:
Sorry? Apparently it may without aggregation.